product_sticker/models/product_sticker.py

```diff
@@ -206,7 +206,12 @@ def _get_stickers(self, products, extra_domain=None):
         no_variant_attribute_lines = product_templates.attribute_line_ids.filtered(
             lambda al: al.attribute_id.create_variant == "no_variant"
         )
-        pp_pavs = products.product_template_variant_value_ids.product_attribute_value_id
+        pp_pavs = self.env["product.attribute.value"].browse()
+        for product in products:
+            if product.product_template_variant_value_ids:
+                pp_pavs |= product.product_template_variant_value_ids.product_attribute_value_id  # noqa: E501
```

Comment:
Binary union tends to be less efficient than regular agreggation in my experience
```suggestion
                pp_pavs += product.product_template_variant_value_ids.product_attribute_value_id  # noqa: E501
```

Reply:
However, in this case, if another product has the same attribute value, sum it would create duplicates, right? Thus I think it's better as it is, or with my other suggestion below.

Reply:
As @yajo said, I don't want duplicates. Just a set of recordsets.

Reply:
Ok, makes sense. Just be aware the larger the recordset (thoundans of records) the larger the performance drawback of the union, as an `OrderedSet` is being made every time